presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1279,6 +1290,100 @@ public void revokeTablePrivileges(MetastoreContext metastoreContext, String data
         }
     }
 
+    @Override
+    public long lock(MetastoreContext metastoreContext, String databaseName, String tableName)
+    {
+        try {
+            final LockComponent lockComponent = new LockComponent(LockType.EXCLUSIVE, LockLevel.TABLE, databaseName);
+            lockComponent.setTablename(tableName);
+            final LockRequest lockRequest = new LockRequest(Lists.newArrayList(lockComponent),
+                    metastoreContext.getUsername(),
+                    InetAddress.getLocalHost().getHostName());
+            LockResponse lockResponse = stats.getLock().wrap(() -> getMetastoreClientThenCall(metastoreContext, client -> client.lock(lockRequest))).call();
+            LockState state = lockResponse.getState();
+            long lockId = lockResponse.getLockid();
+            final AtomicBoolean acquired = new AtomicBoolean(state.equals(LockState.ACQUIRED));
+
+            try {
+                if (state.equals(LockState.WAITING)) {
+                    retry()
+                            .maxAttempts(Integer.MAX_VALUE - 100)
```

Comment:
The maxAttempts is from the Iceberg implemention. Since the acquiration of the lock is wrapped by `RetryDriver` to execute, the retries are restricted by `maxRetryTime`(30s default) and `maxAttempts`(10 default), any of which reaches the limit will abort the retries. We only want to limit the retries by the timeout, so it is reasonable to set the maxAttempts to a large count. Also, to avoid any unnecessary boundary issue, the retry is set conservatively as `Integer.MAX_VALUE - 100`.

Reply:
sgtm, thanks for the context!